### PR TITLE
Fix misspells

### PR DESCRIPTION
**Pull request by MisterDA**
Noticed a few misspells in the documentation, ran a checker on the codebase.

### CODE REVIEW

File: docs/markdown/Running-Meson.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ the usual way. A list of backends can be obtained with `meson setup --help`.
 
 Sometimes you want to add extra compiler flags, this can be done by passing
 them in environment variables when calling meson. See [the reference
-tables](Reference-tables.md#compiler-and-linker-flag-envrionment-variables) for
+tables](Reference-tables.md#compiler-and-linker-flag-environment-variables) for
```
Comment (eli-schwartz) [inline docs/markdown/Running-Meson.md:98]: This is even a broken link in the production docs! Good catch.

File: mesonbuild/mesondata.py [hunk 2]
```diff
@@ -282,7 +282,7 @@
 
 macro(meson_ps_disabled_function)
   message(WARNING "The function '${ARGV0}' is disabled in the context of CMake subporjects.\n"
-                  "This should not be an issue but may lead to compilaton errors.")
+                  "This should not be an issue but may lead to compilation errors.")
```
Comment (eli-schwartz) [inline mesonbuild/mesondata.py:285]: Needs to be fixed in the master copy at https://github.com/mesonbuild/meson/blob/e6a167ce092a36017a4ff56b3fb045be62377a25/mesonbuild/cmake/data/preload.cmake#L19

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: mesonbuild/mesondata.py (hunk 2 above)
Comment (MisterDA) [reply mesonbuild/mesondata.py:285]: This is already fixed, you may have picked the wrong commit.
https://github.com/mesonbuild/meson/blob/57b3ef43ed9c81fa18fa17bb4488a119e632eecf/mesonbuild/cmake/data/preload.cmake#L19

Comment (eli-schwartz) [reply mesonbuild/mesondata.py:285]: Sorry, I mis-analyzed the cause of the CI failure (and I'm not sure why I didn't notice that change in the commit diff).

As the CI notes, you need to edit the master copy of the data file first, then run `./tools/gen_data.py` to update mesondata.py -- but it does more than copy over the file. Checking out your PR, then running the tool, adds the following commit diff on top of your changes:

```diff
diff --git a/mesonbuild/mesondata.py b/mesonbuild/mesondata.py
index b1efcbed3..110103e7b 100644
--- a/mesonbuild/mesondata.py
+++ b/mesonbuild/mesondata.py
@@ -1,4 +1,4 @@
-# Copyright 2020 The Meson development team
+# Copyright 2021 The Meson development team
 
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -388,7 +388,7 @@ mesondata = {
     ),
     'cmake/data/preload.cmake': DataFile(
         Path('cmake/data/preload.cmake'),
-        '2b4e632aeb74acb2b441880cf85c0b6fcab03e75b182d3077715a97e739a7918',
+        '7b90f6a60e3b53ecb3a9a206bcdb0e077b9b28452c54fc6cf1295f627d1dabc7',
         file_3_data_preload_cmake,
     ),
 }
```

File: mesonbuild/mesondata.py [hunk 3]
```diff
@@ -30,7 +30,126 @@
 # BEGIN Data section #
 ######################
 
-file_0_data_CMakeListsLLVM_txt = '''\
+file_0_data_preload_cmake = '''\
```
Comment (eli-schwartz) [inline mesonbuild/mesondata.py:33]: Ouch! This is very messy to review. @mensinda this should be sorted...

Comment (MisterDA) [reply mesonbuild/mesondata.py:33]: Ok, I'll relaunch the tool and rebase when this one is merged :)

Comment (eli-schwartz) [reply mesonbuild/mesondata.py:33]: You can rebase now.

**Comment by MisterDA** [comment]
There's still a big strange update to mesondata.py. Is it just `./tools/gen_data.py`?

**Comment by eli-schwartz** [comment]
#8197

**Comment by jpakkane** [comment]
Some things have landed in master, so this requires one more rebase. I'll merge this once that is clear.

**Comment by MisterDA** [comment]
Top! rebased, it's super clean now

**Review by eli-schwartz — APPROVED** [review]

**Pull request merged**